File: src/main/perl/PerlProgramming/InsertionSort.pl
#!/usr/bin/env perl -w

use strict;

my @arr=(1,2,3,4,5,6,1,3,6);

my $i = 0;
foreach (@arr) {
	my $j = $i;

	while( $j > 0 && $arr[$j] < $arr[$j-1] ) {
		my $temp = $arr[$j];
		$arr[$j] = $arr[$j - 1];
		$arr[$j - 1] = $temp;
		$j = $j - 1;
	}
	$i = $i + 1;
}

$i = 0;
foreach ( @arr ) {

	print($arr[$i] ." ");
	$i = $i + 1;

}


1;

File: src/main/python/basic/insertion_sort.py
#!/usr/bin/python


def insertion_sort(arr):
    n = len(arr)
    for i in range(1, n):
        key = arr[i]
        j = i - 1
        while j >= 0 and arr[j] > key:
            arr[j + 1] = arr[j]
            j = j - 1
        arr[j + 1] = key


if __name__ == '__main__':
    array = [3, 6, 2, 12, 56, 8]
    insertion_sort(array)
    print(array)